src/Evolution/DiscontinuousGalerkin/Messages/BoundaryMessage.hpp

```diff
@@ -0,0 +1,117 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <ostream>
+#include <type_traits>
+
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Time/TimeStepId.hpp"
+#include "Utilities/PrettyType.hpp"
+
+#include "Evolution/DiscontinuousGalerkin/Messages/BoundaryMessage.decl.h"
+
+namespace evolution::dg {
+/*!
+ * \brief [Charm++ Message]
+ * (https://charm.readthedocs.io/en/latest/charm%2B%2B/manual.html#messages)
+ * intended to be used in `receive_data` calls on the elements to send boundary
+ * data from one element on one node, to a different element on a (potentially)
+ * different node.
+ *
+ * If this message is to be sent across nodes, the `pack()` and `unpack()`
+ * methods will be called on the sending and receiving node, respectively.
+ */
+template <size_t Dim>
+struct BoundaryMessage : public CMessage_BoundaryMessage<Dim> {
+  using base = CMessage_BoundaryMessage<Dim>;
+
+  size_t subcell_ghost_data_size;
+  size_t dg_flux_data_size;
+  bool sent_across_nodes;
+  ::TimeStepId current_time_step_id;
+  ::TimeStepId next_time_step_id;
+  Mesh<Dim> volume_or_ghost_mesh;
+  Mesh<Dim - 1> interface_mesh;
+
+  // If set to nullptr then we aren't sending that type of data.
+  double* subcell_ghost_data;
+  double* dg_flux_data;
+
+  BoundaryMessage() = default;
+
+  BoundaryMessage(const size_t subcell_ghost_data_size_in,
+                  const size_t dg_flux_data_size_in,
+                  const bool sent_across_nodes_in,
+                  const ::TimeStepId& current_time_step_id_in,
+                  const ::TimeStepId& next_time_step_id_in,
+                  const Mesh<Dim>& volume_or_ghost_mesh_in,
+                  const Mesh<Dim - 1>& interface_mesh_in,
+                  double* subcell_ghost_data_in, double* dg_flux_data_in);
+
+  /*!
+   * \brief This is the size necessary to allocate a BoundaryMessage with only
+   * the member variables (no actual subcell/dg data, only pointers).
+   *
+   * This evaluates to 240 for Dim == 1, 264 for Dim == 2, and 296 for Dim == 3.
+   */
+  static size_t total_size_without_data();
```

Comment:
maybe `total_number_of_bytes` to make it clear the unit of the size? Same for next function